extension/pprofextension/factory.go

```diff
@@ -44,7 +44,7 @@ func (f *Factory) CreateDefaultConfig() configmodels.Extension {
 			TypeVal: typeStr,
 			NameVal: typeStr,
 		},
-		Endpoint: "localhost:1777",
+		Endpoint: "0.0.0.0:1777",
```

Comment:
While I understand making receivers default to 0.0.0.0 I'm not sure pprof should as it's intended as a diagnostic tool. Usually you would ssh to the box (or through some proxy) to access it. There could also be security implications to making it unprotected by default. Curious what others think.

Another reason it differs from receivers is that receivers are passive and simply accept incoming data, whereas pprof and zpages return data (that may or may not have some sensitive things in it).

Reply:
For these two I would err on the side of caution and make the user explicitly opt-in to exposing it beyond localhost. @bogdandrutu @tigrannajaryan any opinion?

Reply:
I think @jrcamp has a good point. Unless we know of use cases where remote access to pprof and zpages is needed it is best to keep them on localhost by default.